lib/nameable.rb

```diff
@@ -0,0 +1,30 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+  end
+end
+
+class BaseDecorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    nameable.correct_name[0...10] unless nameable.correct_name.length < 10
+  end
+end
```

Comment:
- [ OPTIONAL ] Great job working on the functionality so far but I think it would be amazing if you added each class in a separate file to make your file base more organized and clear to follow 🎉💯🙏🏻so I highly recommend doing that ⭐✨

Reply:
Thank you @Reem-lab, I'll take notes and apply the suggested chagnes!